src/services/athena/createQuerySql.ts

```diff
@@ -0,0 +1,93 @@
+import { CreateQuerySqlResult } from '../../types/createQuerySqlResult'
+import {
+  DataRequestParams,
+  IdentifierTypes
+} from '../../types/dataRequestParams'
+import { getEnv } from '../../utils/helpers'
+
+export const createQuerySql = (
+  requestData: DataRequestParams
+): CreateQuerySqlResult => {
+  console.info('Generating Athena SQL query string')
+  const identifierType = requestData.identifierType
+  const identifiers = getIdentifiers(identifierType, requestData)
+
+  if (identifiers.length < 1) {
+    return {
+      sqlGenerated: false,
+      error: `No ids of type: ${identifierType}`
+    }
+  }
+
+  if (!requestData.dataPaths || requestData.dataPaths.length < 1) {
+    return {
+      sqlGenerated: false,
+      error: 'No dataPaths in request'
+    }
+  }
+
+  const sqlSelectStatement = formatSelectStatement(requestData.dataPaths)
+
+  const sqlWhereStatement = formatWhereStatment(identifierType, identifiers)
+
+  const dataSource = `${getEnv('ATHENA_DATABASE_NAME')}.${getEnv(
+    'ATHENA_TABLE_NAME'
+  )}`
+
+  const queryString = `SELECT ${sqlSelectStatement} FROM ${dataSource} WHERE ${sqlWhereStatement}`
```

Comment:
Is it possible to do parameterised queries in Athena? Just thinking, this looks it could be a bit Bobby Tables https://xkcd.com/327/?

Reply:
Yes, this is a good point and one I had some concern about. At present Athena only really allows parameterisation of the 'WHERE' parts, but there may be another way using what are termed Prepared statements - I'll investigate further.

Reply:
Prepared Statements have the same limitation (i.e. only allowing parameterisation of the WHERE clause). But this is something that should be addressed as it is a risk.

Reply:
Athena only supports parameterisation of the WHERE clause -  I have updated the  function to return a parameterised query to the extent supported.  The other values are all validated at various points - piiTypes and indentifierTypes are defined types and the dataPaths are validated for format.